src/python/pants/core/goals/fmt.py

```diff
@@ -130,23 +143,80 @@ def files(self) -> tuple[str, ...]:
             return self.elements
 
     @classmethod
-    def _get_registration_rules(cls) -> Iterable[UnionRule]:
-        yield from super()._get_registration_rules()
+    def _get_registration_rules(cls, *, partitioner_type: PartitionerType) -> Iterable[UnionRule]:
```

Comment:
It's a bit odd for this to be a parameter for the method rather than a `ClassVar`, since you could _technically_ call the method multiple times with different arguments, and end up with conflicting rules installed...?

Reply:
Fair point. I was gonna rename the method, so I'll do that too in the refactoring pr